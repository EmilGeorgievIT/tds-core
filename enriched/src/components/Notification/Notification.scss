@import '~telus-thorium-core/scss/settings/variables';
@import '~telus-thorium-core/scss/utility/mixins';

.notification {
  padding: 22px 37px;
  background-color: $color-athens-grey;
  display: table;
  width: 100%;
  min-height: 68px;

  @include from-breakpoint(small) {
    padding: 22px 45px;
  }

  > .notification__content {
    display: table-cell;
    vertical-align: middle;
    width: 100%;
  }

  > .notification__content > *:first-child {
    margin-top: 0;
  }

  > .notification__content > *:last-child {
    margin-bottom: 0;
  }
}

.notification--instructional {
  background-color: $color-athens-grey;
}

.notification--branded {
  background-color: $color-white-lilac;
}

.notification--success {
  background-color: $color-panache;
}

.notification--error {
  background-color: $color-lavender-blush;
}
