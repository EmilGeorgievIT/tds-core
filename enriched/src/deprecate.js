/* eslint-disable import/prefer-default-export */

export const warn = (componentName, message) => {
  if (process.env.NODE_ENV === 'production') {
    return;
  }

  console.log(`[TDS] [Deprecate] ${componentName}: ${message}`); // eslint-disable-line no-console
};
