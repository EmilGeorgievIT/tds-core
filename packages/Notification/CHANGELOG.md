# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.0.1"></a>
## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-notification@1.0.0...@tds/core-notification@1.0.1) (2018-03-22)




**Note:** Version bump only for package @tds/core-notification
