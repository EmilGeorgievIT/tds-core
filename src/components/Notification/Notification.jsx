import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../utils/safeRest'

import Grid from '../../old-components/Grid/Grid'
import DecorativeIcon from '../Icons/DecorativeIcon/DecorativeIcon'
import ColoredTextProvider from '../Typography/ColoredTextProvider/ColoredTextProvider'
import Paragraph from '../Typography/Paragraph/Paragraph'
import Flexbox from '../Flexbox/Flexbox'
import Box from '../Box/Box'

import messagingStyles from '../Messaging.modules.scss'
import iconWrapperStyles from '../Icons/IconWrapper.modules.scss'
import styles from './Notification.modules.scss'

const {Container, Row, Column} = Grid

const iconByVariant = {
  success: {
    symbol: 'checkmark',
    color: 'primary',
  },
  error: {
    symbol: 'exclamationPointCircle',
    color: 'error',
  },
}

const isImportant = variant => variant === 'success' || variant === 'error'

const renderIcon = icon => (
  <Box right={3} dangerouslyAddClassName={iconWrapperStyles.fixLineHeight}>
    <DecorativeIcon symbol={icon.symbol} variant={icon.color} />
  </Box>
)

const renderContent = (variant, children) => {
  const content = <Paragraph bold={isImportant(variant)}>{children}</Paragraph>

  if (variant === 'error') {
    return (
      <ColoredTextProvider colorClassName={messagingStyles.errorText}>
        {content}
      </ColoredTextProvider>
    )
  }

  return content
}

/**
 * A banner that highlights important messages.
 */
const Notification = ({variant, children, ...rest}) => (
  <Box {...safeRest(rest)} vertical={3} dangerouslyAddClassName={styles[variant]}>
    <Container limitWidth>
      <Row>
        <Column xs={12}>
          <Flexbox direction="row">
            {isImportant(variant) ? renderIcon(iconByVariant[variant]) : undefined}

            {renderContent(variant, children)}
          </Flexbox>
        </Column>
      </Row>
    </Container>
  </Box>
)

Notification.propTypes = {
  /**
   * The appearance.
   */
  variant: PropTypes.oneOf(['instructional', 'branded', 'success', 'error']),
  /**
   * The message. Can be raw text or text components.
   */
  children: PropTypes.node.isRequired,
}

Notification.defaultProps = {
  variant: 'instructional',
}

export default Notification
