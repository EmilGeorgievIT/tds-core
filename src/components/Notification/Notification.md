## Minimal usage

```
<Notification>Login to modify your account preferences.</Notification>
```

## Variants

Use the `variant` prop to alter the Notification's appearance.


### Branded

```
<Notification variant="branded">Welcome to TELUS.</Notification>
```

### Success

```
<Notification variant="success">Thank you for registering.</Notification>
```

### Error

```
<Notification variant="error">
  <p>Looks like our registration system is temporarily down.</p>
  <p>Come back another time to register for My Account.</p>
</Notification>
```

### Instructional

```
<Notification variant="instructional">
  Fill out the form below, then click <em>"Next"</em> to continue.
</Notification>
```
