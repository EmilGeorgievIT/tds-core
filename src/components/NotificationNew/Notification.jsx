import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../safeRest'

import Container from '../Grid/Container/Container'
import Icon from '../Icon/Icon'
import ColoredTextProvider from '../Typography/ColoredTextProvider/ColoredTextProvider'
import Paragraph from '../Typography/Paragraph/Paragraph'

import styles from './Notification.modules.scss'

const iconByVariant = {
  success: 'checkmark',
  error: 'exclamation-point-circle'
}

const isImportant = variant => variant === 'success' || variant === 'error'

const renderIcon = glyph => (
  <span className={styles.icon}>
    <Icon glyph={glyph} aria-hidden="true" />
  </span>
)

const renderContent = (variant, children) => {
  const content = (
    <Paragraph bold={isImportant(variant)}>
      {children}
    </Paragraph>
  )

  if (variant === 'error') {
    return (
      <ColoredTextProvider colorClassName={styles.errorText}>
        {content}
      </ColoredTextProvider>
    )
  }

  return content
}

/**
 * A banner that highlights important messages.
 */
const Notification = ({ variant, children, ...rest }) => (
  <div {...safeRest(rest)} className={styles[variant]}>
    <Container limitWidth>
      <div className={styles.flexRow}>
        {isImportant(variant) ? renderIcon(iconByVariant[variant]) : undefined}

        {renderContent(variant, children)}
      </div>
    </Container>
  </div>
)

Notification.propTypes = {
  /**
   * The appearance.
   */
  variant: PropTypes.oneOf([
    'instructional',
    'branded',
    'success',
    'error'
  ]),
  /**
   * The message. Can be raw text or text components.
   */
  children: PropTypes.node.isRequired
}

Notification.defaultProps = {
  variant: 'instructional'
}

export default Notification
