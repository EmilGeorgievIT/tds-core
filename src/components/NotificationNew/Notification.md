## Minimal usage

Used for all general or instructional type messages. Defaults to instructional.

Only pass in text. Write in complete sentences.

```
<Notification>
  <strong>Tip:</strong> The services are best suited for larger business organizations ordering more than 50 plans on one account.
</Notification>
```

## Variants

Use the `variant` prop to alter the Notification's appearance.


### Branded

Used for feedback or chat related messages.

```
<Notification variant="branded">
  <strong>Tell us what you think</strong> It’s in our nature to listen. As TELUS.com continues to evolve, we’d love to hear more from you.
</Notification>
```

### Success

Used for successful transactional messages. Text is bolded to show importance.

```
<Notification variant="success">
  Your password has been successfully changed.
</Notification>
```

### Error

Used for error messages. Text is bolded to show importance.

```
<Notification variant="error">
  Looks like our registration system is temporarily down. You’ll need to come back another time to register for My Account. In the meantime, return to TELUS.com.
</Notification>
```
