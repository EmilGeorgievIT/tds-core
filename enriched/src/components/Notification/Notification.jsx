import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { warn } from '../../deprecate';

if (process.env.BROWSER) {
  require('./Notification.scss');
}

const ErrorIcon = () => (
  <i className="icon icon-core-exclamation-point-circle c-notification__icon" />
);

const Notification = ({ className, variant, children, ...rest }) => {
  if (className) {
    warn('Notification', 'Custom CSS classes are deprecated. This component does not support custom styling.');
  }

  if (rest.style) {
    warn('Notification', 'Inline styles are deprecated. This component does not support custom styling.');
  }

  const classes = classNames('notification', className, {
    [`notification--${variant}`]: variant
  });

  return (
    <header role="banner" {...rest} className={classes}>
      <div className="notification__content">
        {variant === 'error' && <ErrorIcon />}

        <div className="c-notification__text">
          {children}
        </div>
      </div>
    </header>
  );
};

Notification.propTypes = {
  className: PropTypes.string,
  variant: PropTypes.oneOf([
    'instructional',
    'branded',
    'success',
    'error'
  ]),
  children: PropTypes.node.isRequired
};

export default Notification;
