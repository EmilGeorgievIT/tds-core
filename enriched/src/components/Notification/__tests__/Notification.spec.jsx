import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import sinon from 'sinon';

import * as deprecate from '../../../deprecate';
import Notification from '../../Notification/Notification';

describe('<Notification />', () => {
  const doShallow = ({ ...overrides }) => (
    shallow(<Notification {...overrides}>Some content</Notification>)
  );

  beforeEach(() => {
    sinon.spy(deprecate, "warn");
  });

  afterEach(() => {
    deprecate.warn.restore();
  });

  it('renders', () => {
    const notification = doShallow();

    expect(toJson(notification)).toMatchSnapshot();
  });

  it('can have a variant', () => {
    let notification = doShallow({ variant: undefined });
    expect(notification).not.toHaveClassName('notification--');

    notification = doShallow({ variant: 'success' });
    expect(notification).toHaveClassName('notification notification--success');
  });

  it('adds icon for error variant', () => {
    // Todo: Use render when it becomes usable with jest
    // See https://github.com/blainekasten/enzyme-matchers/issues/25
    const notification = doShallow({ variant: 'error' });

    expect(notification.find('ErrorIcon')).toBePresent();
  });

  it('passes attributes to DOM node', () => {
    const notification = doShallow({ id: 'hello', title: 'my title' });

    expect(notification).toHaveProp('id', 'hello');
    expect(notification).toHaveProp('title', 'my title');
  });

  it('accepts but deprecates custom classes', () => {
    const notification = doShallow({ className: 'some-class' });

    expect(notification).toHaveClassName('some-class');
    expect(deprecate.warn.called).toBeTruthy();
  });

  it('accepts but deprecates inline styles', () => {
    const style = { color: 'blue' };
    const notification = doShallow({ style });

    expect(notification).toHaveProp('style', style);
    expect(deprecate.warn.called).toBeTruthy();
  });
});
