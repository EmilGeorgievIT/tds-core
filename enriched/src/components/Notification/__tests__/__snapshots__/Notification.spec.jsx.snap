// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> renders 1`] = `
<div
  className="notification"
>
  <div
    className="notification__content"
  >
    <div
      className="c-notification__text"
    >
      Some content
    </div>
  </div>
</div>
`;
